### ISSUE TITLE
Cannot choose channel when creating rule to trigger on Thing Event

### ISSUE THREAD

**Issue by kvolden** [openhab/openhab-webui#1996]
## The problem

I have a fresh install of Openhabian, running Openhab 4.0.0 on a raspberry pi. I want to create a rule to trigger on a thing event. When creating the rule in the UI, I choose "Thing Event". As the thing I choose "Local Sun (astro:sun:local)", which I know for a fact produces events that I can also see at expected intervals in the events log. But when I then click on "channel" to choose the correct one, nothing happens. It doesn’t show a new dialogue, it’s just dead.

## Expected behavior

A dialogue should open showing the different thing event channels to choose from.

## Steps to reproduce

Click: Settings -> Rules -> (+) -> Add Trigger -> Thing Event -> Choose thing with events (Local Sun in my case) -> Channel

## Your environment

```yaml
runtimeInfo:
  version: 4.0.0
  buildString: Release Build
locale: en-NO
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 17.0.7
  javaVendor: Raspbian
  osName: Linux
  osVersion: 6.1.21-v7+
  osArchitecture: arm
  availableProcessors: 4
  freeMemory: 75349472
  totalMemory: 204009472
  startLevel: 100
bindings: null
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: light
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 6
    language: nb-NO
    languages:
      - nb-NO
      - nb
      - no
      - nn
      - en-US
      - en
      - sv
      - da
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/114.0.0.0 Safari/537.36
timestamp: 2023-07-28T16:38:44.389Z
```

## Browser console

```txt
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'filter')
    at 20.app.c6e04ce7588be7e4ae8a.js:1:158388
    at Array.map (<anonymous>)
    at 20.app.c6e04ce7588be7e4ae8a.js:1:158313
(anonymous) @ 20.app.c6e04ce7588be7e4ae8a.js:1
(anonymous) @ 20.app.c6e04ce7588be7e4ae8a.js:1
Promise.then (async)
created @ 20.app.c6e04ce7588be7e4ae8a.js:1
Ue @ app.c6e04ce7588be7e4ae8a.js:7
tn @ app.c6e04ce7588be7e4ae8a.js:7
e._init @ app.c6e04ce7588be7e4ae8a.js:7
i @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
init @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
d @ app.c6e04ce7588be7e4ae8a.js:7
f @ app.c6e04ce7588be7e4ae8a.js:7
d @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
e._update @ app.c6e04ce7588be7e4ae8a.js:7
a @ app.c6e04ce7588be7e4ae8a.js:7
fn.get @ app.c6e04ce7588be7e4ae8a.js:7
fn @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
Sn.$mount @ app.c6e04ce7588be7e4ae8a.js:7
Sn.$mount @ app.c6e04ce7588be7e4ae8a.js:7
init @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
d @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
O @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
e._update @ app.c6e04ce7588be7e4ae8a.js:7
a @ app.c6e04ce7588be7e4ae8a.js:7
fn.get @ app.c6e04ce7588be7e4ae8a.js:7
fn.run @ app.c6e04ce7588be7e4ae8a.js:7
pn @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
Ze @ app.c6e04ce7588be7e4ae8a.js:7
Promise.then (async)
Ye @ app.c6e04ce7588be7e4ae8a.js:7
at @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
fn.update @ app.c6e04ce7588be7e4ae8a.js:7
pe.notify @ app.c6e04ce7588be7e4ae8a.js:7
set @ app.c6e04ce7588be7e4ae8a.js:7
mn.set @ app.c6e04ce7588be7e4ae8a.js:7
chooseThingCategory @ 20.app.c6e04ce7588be7e4ae8a.js:1
Ue @ app.c6e04ce7588be7e4ae8a.js:7
n @ app.c6e04ce7588be7e4ae8a.js:7
Ue @ app.c6e04ce7588be7e4ae8a.js:7
e.$emit @ app.c6e04ce7588be7e4ae8a.js:7
(anonymous) @ app.c6e04ce7588be7e4ae8a.js:7
t.a @ app.c6e04ce7588be7e4ae8a.js:7
dispatchEvent @ app.c6e04ce7588be7e4ae8a.js:7
onClick @ app.c6e04ce7588be7e4ae8a.js:7
```

## Additional information

I tried to work around this by manually adding the channelUID to the rule in the code tab, but it fails to trigger from the event, even though I have verified the event itself triggered from the logs. I'm not sure if this is a symptom of the same problem, or a different one.

**Comment by Alext73** [comment]
i had the same problem.

**Comment by lolodomo** [comment]
Would be cool to have e fix for 4.0.2.

**Comment by florian-h05** [comment]
@lolodomo 
I am waiting for an answer at https://github.com/openhab/openhab-core/pull/3335#discussion_r1278566388.
The issue itself is easy to fix, but I need that answer.
